src/main/java/com/amazonaws/services/dynamodbv2/LockItem.java

```diff
@@ -38,7 +39,7 @@
     private final String partitionKey;
     private final Optional<String> sortKey;
 
-    private final Optional<ByteBuffer> data;
+    private final AtomicReference<ByteBuffer> data;
```

Comment:
Does it need to be an AtomicReference? Not all fields in this class are atomic variables (e.g. recordVersionNumber)

Reply:
No it doesn't. All updates to it are within a synchronized block. Will change.